config/onprem-iso-config.ign

```diff
@@ -58,7 +66,12 @@
         "name": "assisted-service-certs.service"
       },
       {
-        "contents": "[Unit]\nAfter=network-online.target\n\n[Service]\nType=forking\nRestart=no\nExecStart=podman pod create --name assisted-installer -p 8443\n\n[Install]\nWantedBy=multi-user.target\n",
+        "contents": "[Unit]\nAfter=network-online.target\n\n[Service]\nType=forking\nRestart=no\nExecStart=/bin/bash /etc/assisted-service/startup_script.sh\n\n[Install]\nWantedBy=multi-user.target\n",
+        "enabled": true,
+        "name": "assisted-service-ip-configuration.service"
+      },
+      {
+        "contents": "[Unit]\nAfter=assisted-service-ip-configuration.service\n\n[Service]\nType=forking\nRestart=no\nExecStart=podman pod create --name assisted-installer -p 8443\n\n[Install]\nWantedBy=multi-user.target\n",
```

Comment:
I think this actually creates a race condition. 
`After=` only actually guarantees that this unit will start after the specified one is started. There is no guarantee that we will finish writing out the environment file in `assisted-service-ip-configuration` before this unit is started.

That said it's hard to follow the flow here as I have to decode all of the script content. This is what lead to my other comment :smile:

Reply:
Nope, was confusing `After` with `Wants` ... :tired_face: